src/shared/apis/queryCache.ts

```diff
@@ -120,10 +145,21 @@ class QueryCache {
     values: RunQueryResult
   ): void => {
     event('Query Cache was Set', {context: 'queryCache', queryID})
-    this.cache[queryID] = {
-      ...this.initializeCacheByID(queryID, hashedVariables),
-      dateSet: Date.now(),
-      values,
+    if (values.type === 'SUCCESS') {
```

Comment:
i'm not normally one to say something here, but the duplication in these conditions rubs me the wrong way for some reason

Reply:
Sounds good